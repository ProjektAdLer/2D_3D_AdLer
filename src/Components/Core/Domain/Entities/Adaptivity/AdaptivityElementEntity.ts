import { AdaptivityElementQuestionDifficultyTypes } from "../../Types/Adaptivity/AdaptivityElementQuestionDifficultyTypes";
import LearningElementEntity from "../LearningElementEntity";

export default class AdaptivityElementEntity {
  element: LearningElementEntity;
  introText: string;
  shuffleTask: boolean;
  tasks: AdaptivityTaskEntity[];
}

export class AdaptivityTaskEntity {
  taskId: number;
  taskTitle: string;
  taskOptional: boolean;
  requieredDifficulty: AdaptivityElementQuestionDifficultyTypes;
  questions: AdaptivityQuestionEntity[];
}

export class AdaptivityQuestionEntity {
  questionType: string;
  questionId: number;
  questionDifficulty: AdaptivityElementQuestionDifficultyTypes;
  questionText: string;
  questionAnswers: AdaptivityAnswerEntity[];
  trigger: AdaptivityTriggerEntity[];
}

export class AdaptivityAnswerEntity {
  answerId: number;
  answerText?: string;
  answerImage?: string;
}

export class AdaptivityTriggerEntity {
  triggerId: number;
  triggerType: string;
  triggerCondition: string;
  triggerAction: AdaptivityActionEntity;
}

export class AdaptivityActionEntity {
  actionType: string;
  actionData: string | number;
}
