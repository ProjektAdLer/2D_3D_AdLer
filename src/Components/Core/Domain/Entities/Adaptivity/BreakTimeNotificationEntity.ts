export enum BreakTimeNotificationType {
  None = 0,
  Short = 5,
  Medium = 15,
  Long = 30,
}

export default class BreakTimeNotificationEntity {
  breakTimeIntervalCounter: number;
}
