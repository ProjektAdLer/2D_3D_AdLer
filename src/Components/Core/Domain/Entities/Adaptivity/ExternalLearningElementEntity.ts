import { ComponentID } from "../../Types/EntityTypes";
import { LearningElementTypeStrings } from "../../Types/LearningElementTypes";

export default class ExternalLearningElementEntity {
  id: ComponentID;
  worldID: ComponentID;
  name: string;
  type: LearningElementTypeStrings;
}
