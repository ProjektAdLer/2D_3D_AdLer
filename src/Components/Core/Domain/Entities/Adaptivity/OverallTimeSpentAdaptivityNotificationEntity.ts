export enum OverallTimeSpentAdaptivityNotificationBreakType {
  None = 0,
  Short = 30,
  Medium = 60,
  Long = 90,
}

export enum OverallTimeSpentAdaptivityNotificationWaitingType {
  Short = 5,
  Medium = 15,
  Long = 30,
}

export default class OverallTimeSpentAdaptivityNotificationEntity {
  notificationType: OverallTimeSpentAdaptivityNotificationBreakType;
}
