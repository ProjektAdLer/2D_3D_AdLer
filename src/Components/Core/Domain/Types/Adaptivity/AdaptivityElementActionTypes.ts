// TODO: add string representations
export enum AdaptivityElementActionTypes {
  CommentAction = "CommentAction",
  ReferenceAction = "AdaptivityReferenceAction",
  ContentAction = "AdaptivityContentReferenceAction",
}
