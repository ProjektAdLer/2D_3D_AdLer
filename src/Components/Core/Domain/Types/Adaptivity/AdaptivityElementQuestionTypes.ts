export enum AdaptivityElementQuestionTypes {
  singleChoice = "SingleChoice",
  multipleChoice = "MultipleChoice",
}
