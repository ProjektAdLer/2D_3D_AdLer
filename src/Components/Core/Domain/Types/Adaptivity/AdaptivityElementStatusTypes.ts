export enum AdaptivityElementStatusTypes {
  correct = "Correct",
  incorrect = "Incorrect",
}
