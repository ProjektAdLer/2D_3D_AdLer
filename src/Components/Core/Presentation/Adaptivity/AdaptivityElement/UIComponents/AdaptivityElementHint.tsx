import { AdaptivityElementActionTypes } from "src/Components/Core/Domain/Types/Adaptivity/AdaptivityElementActionTypes";
import { AdaptivityHint } from "../AdaptivityElementViewModel";
import { useEffect } from "react";
import { useTranslation } from "react-i18next";
import StyledButton from "~ReactComponents/ReactRelated/ReactBaseComponents/StyledButton";

export default function AdaptivityElementHint({
  hint,
  setHeaderText,
  closeHint,
}: {
  hint: AdaptivityHint;
  setHeaderText: (headerText: string) => void;
  closeHint: () => void;
}) {
  const { t: translate } = useTranslation("learningElement");

  useEffect(() => {
    setHeaderText(translate("hintTitle"));
  }, [setHeaderText, translate]);

  return (
    <>
      {hint.hintAction.hintActionType ===
        AdaptivityElementActionTypes.CommentAction &&
        hint.hintAction.textData !== undefined && (
          <div className="flex items-center justify-center p-2 pl-4 my-4 bg-buttonbgblue rounded-xl">
            {hint.hintAction.textData}
          </div>
        )}

      {hint.hintAction.hintActionType ===
        AdaptivityElementActionTypes.ContentAction && null}

      {hint.hintAction.hintActionType ===
        AdaptivityElementActionTypes.ReferenceAction &&
        hint.hintAction.textData !== undefined && (
          <div>{hint.hintAction.textData}</div>
        )}
      <div className="flex justify-end w-full">
        <StyledButton shape="freefloatcenter" onClick={closeHint}>
          <p className="text-sm">{translate("nextButton")}</p>
        </StyledButton>
      </div>
    </>
  );
}
