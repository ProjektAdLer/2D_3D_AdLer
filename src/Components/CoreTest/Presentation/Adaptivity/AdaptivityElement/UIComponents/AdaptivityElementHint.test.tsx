import { render } from "@testing-library/react";
import React from "react";
import AdaptivityElementHint from "../../../../../Core/Presentation/Adaptivity/AdaptivityElement/UIComponents/AdaptivityElementHint";
import { AdaptivityElementActionTypes } from "../../../../../Core/Domain/Types/Adaptivity/AdaptivityElementActionTypes";

describe("AdaptivityElementHint", () => {
  // ANF-ID: [EWE0004]
  test("should render CommentAction", () => {
    const { container } = render(
      <AdaptivityElementHint
        hint={{
          hintID: 0,
          hintAction: {
            hintActionType: AdaptivityElementActionTypes.CommentAction,
            textData: "test",
          },
          showOnIsWrong: true,
        }}
        setHeaderText={() => {}}
      />
    );

    expect(container).toMatchSnapshot();
  });

  test("should render ReferenceAction", () => {
    const { container } = render(
      <AdaptivityElementHint
        hint={{
          hintID: 0,
          hintAction: {
            hintActionType: AdaptivityElementActionTypes.ReferenceAction,
            textData: "test",
            idData: 0,
          },
          showOnIsWrong: true,
        }}
        setHeaderText={() => {}}
      />
    );

    expect(container).toMatchSnapshot();
  });

  // ANF-ID: [EWE0044]
  test("should render ContentAction", () => {
    const { container } = render(
      <AdaptivityElementHint
        hint={{
          hintID: 0,
          hintAction: {
            hintActionType: AdaptivityElementActionTypes.ContentAction,
            textData: "test",
            idData: 0,
          },
          showOnIsWrong: true,
        }}
        setHeaderText={() => {}}
      />
    );

    expect(container).toMatchSnapshot();
  });
});
