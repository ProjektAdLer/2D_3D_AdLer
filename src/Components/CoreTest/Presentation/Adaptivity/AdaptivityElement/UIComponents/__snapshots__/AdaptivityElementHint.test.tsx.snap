// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdaptivityElementHint should render CommentAction 1`] = `
<div>
  <div
    class="flex items-center justify-center p-2 pl-4 my-4 bg-buttonbgblue rounded-xl"
  >
    test
  </div>
  <div
    class="flex justify-end w-full"
  >
    <div
      class=""
    >
      <button
        class="px-2 py-1 justify-center flex items-center text-sm rounded-lg hover:cursor-pointer hover:border-adlerdarkblue hover:bg-adleryellow lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue"
      >
        <p
          class="text-sm"
        >
          nextButton
        </p>
      </button>
    </div>
  </div>
</div>
`;

exports[`AdaptivityElementHint should render ContentAction 1`] = `
<div>
  <div
    class="flex items-center justify-center p-2 pl-4 my-4 bg-buttonbgblue rounded-xl"
  >
    test
  </div>
  <div
    class="flex justify-end w-full"
  >
    <div
      class=""
    >
      <button
        class="px-2 py-1 justify-center flex items-center text-sm rounded-lg hover:cursor-pointer hover:border-adlerdarkblue hover:bg-adleryellow lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue"
      >
        <p
          class="text-sm"
        >
          nextButton
        </p>
      </button>
    </div>
  </div>
</div>
`;

exports[`AdaptivityElementHint should render ReferenceAction 1`] = `
<div>
  <div>
    test
  </div>
  <div
    class="flex justify-end w-full"
  >
    <div
      class=""
    >
      <button
        class="px-2 py-1 justify-center flex items-center text-sm rounded-lg hover:cursor-pointer hover:border-adlerdarkblue hover:bg-adleryellow lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue"
      >
        <p
          class="text-sm"
        >
          nextButton
        </p>
      </button>
    </div>
  </div>
</div>
`;
